File: apps/executor/pkg/api/controllers/health.go
package controllers

import (
	"net/http"

	"github.com/labstack/echo/v4"
)

// HealthCheck 			godoc
//
//	@Summary		Health check
//	@Description	Health check
//	@Produce		json
//	@Success		200	{object}	map[string]string
//	@Router			/ [get]
//
//	@id				HealthCheck
func HealthCheck(c echo.Context) error {
	return c.JSON(http.StatusOK, map[string]string{
		"status":  "ok",
		"version": "0.0.1",
	})
}

File: crates/executor/src/api/controllers/health.rs
// Copyright (c) 2026 Snapflow. All rights reserved.
//
// Snapflow is licensed under the GNU Affero General Public License v3.0.
// You may obtain a copy of the license at https://www.gnu.org/licenses/agpl-3.0.html
//
// SPDX-License-Identifier: AGPL-3.0

use axum::Json;
use serde_json::{Value, json};

#[utoipa::path(
    get,
    path = "/",
    summary = "Health check",
    description = "Health check",
    operation_id = "HealthCheck",
    responses(
        (status = 200, description = "OK", body = Object)
    )
)]
pub async fn health_check() -> Json<Value> {
    Json(json!({
        "status": "ok",
        "version": env!("CARGO_PKG_VERSION")
    }))
}
